WIP: Add new unified resource management

agibsonccc opened the pull request:
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?
Add new unified resource management for all download + cache (test resources, pretrained models,..)
(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

## Quick checklist

The following checklist helps ensure your PR is complete:

- [ ] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [ ] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [ ] Created tests for any significant new code additions.
- [ ] Relevant tests for your changes are passing.

agibsonccc committed:
ADR: Add new unified resource management

treo commented on deeplearning4j/deeplearning4j-zoo/src/main/java/org/deeplearning4j/zoo/util/BaseLabels.java:
```diff
@@ -141,9 +141,10 @@ protected File getResourceFile(){
             try{
-                if(Downloader.checkMD5OfFile(expMD5, localFile)){
+                //empty string means ignore the MD5
+                if(expMD5.isEmpty() || Downloader.checkMD5OfFile(expMD5, localFile)){
```
Why handle it here? Why not put the special casing into the checker? I can see that you've got that case also checked at a different place.

agibsonccc committed:
Address ADR grammar, minor nit for downloader logic

agibsonccc committed:
Restore file exists check, redo tense

agibsonccc committed:
Update BERTGraphTest.java

agibsonccc committed:
will -> is

agibsonccc committed:
Update 0015 - Unified Resource Manager.md

agibsonccc committed:
Update inconsistent deletion check

agibsonccc committed:
Add delete if necessary

agibsonccc committed:
Update BootstrapFromLocal.java

agibsonccc committed:
Refactoring

1. Removes deeplearning4j-common: merges with resources module
2. Removes accidental keras import refactoring
3. Updates old datasetiterators to use new resources api
4. Collapse downloader related clases (strumpf, resource file,..) from nd4j-common in to resources

treo commented on platform-tests/src/test/java/org/eclipse/deeplearning4j/frameworkimport/tensorflow/BERTGraphTest.java:
```diff
@@ -91,7 +91,7 @@ public void testBert(Nd4jBackend backend) throws Exception {
         //Extract
         File f = new File(saveDir, "bert_mrpc_frozen.pb");
-        if(!f.exists() || !Downloader.checkMD5OfFile("93d82bca887625632578df37ea3d3ca5", f)){
+        if(f.exists() && !Downloader.checkMD5OfFile("93d82bca887625632578df37ea3d3ca5", f)){
```
Why don't you use the `deleteIfCorrupted` method here?

agibsonccc committed:
Handle nit, deprecated old model zoo interface

agibsonccc committed:
Update deprecated link, deleteIfCorrupted usgae
